proton-c/src/proactor/libuv.c

```diff
@@ -1179,7 +1179,10 @@ pn_proactor_t *pn_proactor() {
   pn_proactor_t *p = (pn_proactor_t*)calloc(1, sizeof(pn_proactor_t));
   p->collector = pn_collector();
   p->batch.next_event = &proactor_batch_next;
-  if (!p->collector) return NULL;
+  if (!p->collector) {
+    free(p);
```

Comment:
I hope this is enough to free it at this point.

Reply:
Looks good to me, though stylistically I'd move line 1181 after the 'if' statement.

Reply:
I thought of that, but then decided to keep changes to a minimum. Will you move it or should I resubmit the PR? I'd be OK if you just committed it.